virttest/utils_misc.py

```diff
@@ -1718,6 +1739,20 @@ def get_all_node_meminfo(self):
             meminfo[node] = node_meminfo
         return meminfo
 
+    def get_available_nodes(self):
+        """
+        Get node ids available in host
+
+        :return: The ids of node which is available
+        :rtype: builtin.list
+        """
+        node_list = self.online_nodes
+        memory_list = self.get_all_node_meminfo()
+        for nodenum in memory_list:
+            if int(memory_list[nodenum]["MemTotal"]) == 0 and nodenum in node_list:
+                node_list.remove(nodenum)
+        return node_list
+
```

Comment:
@hsmj1412 you can consider this logic to get all nodes with mem, https://github.com/avocado-framework/avocado-vt/pull/1158/commits/b2ed761977efacc5f634c47d791f986f75ac0051 ?
It should be reliable as host provides the info.

Reply:
It looks better than this PR, thanks a lot!